servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java

```diff
@@ -142,6 +142,16 @@ void createReferences() throws NessieNotFoundException {
     String branchName1 = "createReferences_branch1";
     String branchName2 = "createReferences_branch2";
     assertAll(
+        // invalid source ref & null hash
```

Comment:
This will conflict with the addressed review-comment https://github.com/projectnessie/nessie/pull/2083#discussion_r713701618 which will eagerly throw an IAE when `hash` is null for a tag.

Reply:
Better remove this test.

Reply:
the null check introduced in #2083 happens after we're hitting the NPE so it's still valid

Reply:
I suspect, the test will fail with that change ;)

Reply:
rebased onto latest changes and re-running CI. Test is passing locally. As mentioned, the `checkHashOnRef` check (where the NPE is happening) is done before the null check you added in #2083